### PR TITLE
[test] Adding missing partition tests

### PR DESCRIPTION
**Pull request by LucianoPAlmeida**
I just note that `partition(subrange:)` both `BidirectionalCollection`(specialized) and `MutableCollection` overloads.
So this just adds the tests :)

cc @natecook1000 @timvermeulen @kylemacomber 

Replace this paragraph with a description of your changes and rationale. Provide links to an existing issue or external references/discussions, if appropriate.

### Checklist
- [x] I've added at least one test that validates that my change is working, if appropriate
- [x] I've followed the code style of the rest of the project
- [ ] I've read the [Contribution Guidelines](../CONTRIBUTING.md)
- [ ] I've updated the documentation if necessary

### CODE REVIEW

File: Tests/SwiftAlgorithmsTests/PartitionTests.swift [hunk 1]
```diff
@@ -88,4 +88,42 @@ final class PartitionTests: XCTestCase {
       }
     }
   }
+  
+  func testPartitionWithSubrangeBidirectionalCollection() {
+    for length in 10...20 {
+      let a = Array(0..<length)
+      for j in 0..<length {
+        var b = a
+        let partitionRange = 0..<j
```
Comment (natecook1000) [inline Tests/SwiftAlgorithmsTests/PartitionTests.swift:97]: Looks like this is only testing the subrange at the beginning of the collection — could we iterate over `0...j` as well, for the start of the subrange?

Comment (LucianoPAlmeida) [reply Tests/SwiftAlgorithmsTests/PartitionTests.swift:97]: Right, I've updated!
Let me know if that is what you have in mind  :)

Comment (natecook1000) [reply Tests/SwiftAlgorithmsTests/PartitionTests.swift:97]: IIUC I'm looking for one more nested `for` loop inside this one, a `for k in 0...j`, and then `let partitionRange = k..<j` (with hopefully some better variable names).

The issue is that, when `a` has 10 elements, we're partitioning all the subranges starting at `0`, and also now the subranges ending at `length`, below. What we aren't getting is the middle subranges, like `2..<6`.

Comment (LucianoPAlmeida) [reply Tests/SwiftAlgorithmsTests/PartitionTests.swift:97]: Ah I see, makes sense. 
I'm on it, thank you!

Comment (LucianoPAlmeida) [reply Tests/SwiftAlgorithmsTests/PartitionTests.swift]: Done!
Also, adjusted for stable partition tests :)

Comment (LucianoPAlmeida) [reply Tests/SwiftAlgorithmsTests/PartitionTests.swift]: Ping @natecook1000 =]

**Comment by natecook1000** [comment]
@swift-ci Please test

**Review by natecook1000 — APPROVED** [review]

**Pull request merged**